### DISCUSSION TITLE
Decleare PaginateQuery as Object

### DISCUSSION THREAD

**Discussion by vitriol777** [ppetzold/nestjs-paginate#583] [General]
Thanks for this well appreciated lib! 

There is just one thing, i would like to address:
the query Object of Type PaginateQuery is passed through a decorator which deconstructs the url -> i would really enjoy passing something like {limit: 10, searchText: 'hi'} through a post request.
Would be really nice to have.
Thank you for your attention

**Comment by ppetzold** [comment]
should be straight foward. instead using the decorator, you can simply build your own query object by using req body params and pass it to `paginate()`

**Reply by vitriol777** [reply]
Thank you for the quick reply. maybe i did just fail in doing so. i'll try it again.
sorry for wasting your time in this case.

**Comment by vitriol777** [comment]
I am using ngrx store in the frontend and it would be very cool if i could integrate the queries direktly in the effects like that:

something$ = createEffect(this.actions.pipe(map => {
    const query = {
      do something
    }
    this.api.listSomtehing(query).pipe(....you know the drill
}))

in api service:

listSomething(query: PaginationQuery): Observable<Paginated<Something>> {
   this.http.post(`${this.baseUrl}/rest-api/v1/something`, query)
}

this way i can use my api in mutliple contexts without messing with any url segments

**Comment by vitriol777** [comment]
@ppetzold Thanks you - i forgot to type the response in the frontend and therefor didn't see what i was looking for. so never mind and thank you for the fas response and the great work!

**Comment by vitriol777** [comment]
dear mr @ppetzold can you help me solving this problem? I want to filter the table based on this params:
if table.startDateTime === currentDay or
if currentDay is between table.startDateTime and table.endDateTime

i tried it like that:
query.filter = {
 startDateTime: [
                `$eq:${formatedDay}$or:$gt:${formatedDay}$and:filter.endDateTime$lt:${formatedDay}$or:filter.endDateTime$eq:${formatedDay}`
                ]
            }
            
            as i look at it, it is a stupid approach.
            but how can i query through multiple columns with an object?
            
            thanks in advance! greetings from austria

**Reply by vitriol777** [reply]
in other words i want to achive that:
SELECT startDate, endDate
FROM appointment
WHERE '2023-05-02' between startDateTime and endDateTime

**Reply by ppetzold** [reply]
We support iso dates `?filter.createdAt=$gt:2022-12-20T10:00:00.000Z&filter.createdAt=$lt:2022-12-20T12:00:00.000Z`
